dataprofiler/data_readers/graph_data.py

```diff
@@ -173,15 +174,58 @@ def _format_data_networkx(self):
         networkx_graph = nx.Graph()
 
         # read lines from csv
-        csv_as_list = []
-        if not self._checked_header or self._header == "auto":
+        if not self._checked_header or not self._delimiter:
+            delimiter, quotechar = None, None
             data_as_str = data_utils.load_as_str_from_file(
                 self.input_file_path, self.file_encoding
             )
-            self._header = CSVData._guess_header_row(
-                data_as_str, self._delimiter, self._quotechar
+
```

Comment:
this section is what fixes the bug. If graph data was lazily loaded or loaded without options from Is_match, it wouldn't determine these when thrown into the system. This fixes it.